design-doc-template.adoc

```diff
@@ -0,0 +1,39 @@
+= Title
+:author:            Your Name
+:email:             your.email@redhat.com
+:toc:               left
+:icons:             font
+:keywords:          comma,separated,tags
+:idprefix:
+:idseparator:       -
+:issue-base-url:    https://issues.jboss.org
+
+== Overview
+
+== Issue Metadata
+
+=== Issue:
+
+* {issue-base-url}/WFCORE[WFCORE-XXXX]
+
+=== Related Issues:
+
+* {issue-base-url}/WFLY[WFLY-XXXX]
+
+=== Dev Contacts:
+
+* mailto:{email}[{author}]
+
+=== QE Contacts:
+
+=== Affected Projects or Components:
+
+=== Other Interested Projects:
+
+== Requirements
+
+=== Hard Requirements
+
+=== Nice-to-Have Requirements
```

Comment:
Isn't this a bit paradoxical?

Reply:
Perhaps the wording is odd, but oh well. This outlines things that are being considered for the feature as the analysis is being done but which if not practical (technically or in terms of necessary resources) when it comes time to do the development, their absence should not mean the feature has not met its objectives.

These should be resolved one way or the other as soon as possible so people don't waste time (e.g. writing tests) on things that won't happen. But in the initial analysis phase it's useful to note everything being considered while being clear about the must-haves.

Reply:
I agree its useful to jot down ideas being considered which are not hard requirements.

But makes me wonder, what is the difference (i.e. where is the line) between these so-called "nice to have requirements" and "non-requirements" section below? Does this mean "probably won't be implemented" and "obviously out of scope", respectively?

Reply:
More like "Might not be implemented" and "Definitely out of scope".

The distinction between "might not" and "probably won't" isn't a big deal though. The text for the items can say something about probability.

Sometimes a Non-Requirement is because you just know you aren't going to do it (e.g. no time or not worth the effort), so state that up front and save someone asking.  Other times you're just being thorough and stating what might seem obvious to you but surprisingly often isn't obvious to others. 30 secs of writing a non-requirement can save you 10 minutes of writing a response to a question. It's also a get-out-of-jail-free card if someone later comes along and says the missing thing is a Bug.